pkg/api_builder/lib/api_builder.dart

```diff
@@ -84,11 +84,15 @@ abstract class $utilities {
     shelf.Request request,
     T Function(Map<String, dynamic>) fromJson,
   ) async {
-    // TODO: Consider enforcing that requests should have 'Content-Type' set to
-    // 'application/json'. Notice that we should be careful as the CLI client,
-    // might be sending a different Content-Type.
-    final data = await request.readAsString();
     try {
+      // TODO: Consider enforcing that requests should have 'Content-Type' set to
+      // 'application/json'. Notice that we should be careful as the CLI client,
+      // might be sending a different Content-Type.
+      if (request.mimeType != 'application/json') {
```

Comment:
You might get `application/json; charset="utf-8"` or `application/vnd.pub.v2+json`

Reply:
Added `application/vnd.pub.v2+json` to the check.